File: Desarrollo Software-DS/Practica 1/Sesion2/Opcional/Prototipo/Tecnico.rb
require './Trabajo.rb'

class Tecnico

    protected
    def self.clone()
        return Tecnico.new()
    end

    private
    def self.initialize(n,e,dni,trbSist)
        @nombre=n
        @edad=e
        @DNI=dni
        @trabajos=trbSist

    end

    public
    def realizarPuja()
    end


    public
    attr_accessor :trbRealizar, :nombre, :edad, :DNI, :trabajos
end

File: Tercero/Desarrollo Software-DS/Practica 1/Sesion2/Obligatoria/P1S2/src/p1s2/tecnico.java
package p1s2;

import java.util.Random;
import java.util.*;

public abstract class tecnico extends Thread{

	public trabajo trbRealizar;
        String nombre;
        int edad;
        String DNI;
        ArrayList<trabajo> trabajos;
        
 
        
         public tecnico(String n,int e,String dni, ArrayList<trabajo> sist)
        {
            this.nombre=n;
            this.edad=e;
            this.DNI=dni;
            this.trabajos=sist;
            
        }
         
        public trabajo getTrabajo(){
            return this.trbRealizar;
        }
         
        
        
        @Override
        public abstract void run();
        
        
        
}